pkg/controller/vpc_nat_gateway.go

```diff
@@ -246,7 +246,7 @@ func (c *Controller) handleAddOrUpdateVpcNatGw(key string) error {
 
 	// check or create deployment
 	needToCreate := false
-	_, err = c.config.KubeClient.AppsV1().Deployments(c.config.PodNamespace).
+	oldDp, err := c.config.KubeClient.AppsV1().Deployments(c.config.PodNamespace).
```

Comment:
We should make a deepcopy of the oldDp, other wise the `newPodAnnotations[key] = value` operations below will affect the cached value.

Reply:
it's only get  old  cached value from oldDp, how  affect?

Reply:
https://go.dev/play/p/63SXnSpqSNp

Reply:
> https://go.dev/play/p/63SXnSpqSNp

good ，i see

Reply:
But I find  that this client doesn't have caching
![image](https://user-images.githubusercontent.com/81305854/155472205-60fa781b-580b-4b4b-872f-9633dfedb3f3.png)
![image](https://user-images.githubusercontent.com/81305854/155472299-02672faf-e673-482c-afad-44dd239f375b.png)

Reply:
you're right, i didn't notice it